examples/App.cpp

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+namespace {
+
+void setUpTracer(const char* configFilePath)
+{
+    auto configYAML = YAML::LoadFile(configFilePath);
+    auto config = jaegertracing::Config::parse(configYAML);
+    auto tracer = jaegertracing::Tracer::make(
+        "example-service", config, jaegertracing::logging::consoleLogger());
+    opentracing::Tracer::InitGlobal(
+        std::static_pointer_cast<opentracing::Tracer>(tracer));
+}
+
+void tracedSubroutine(const std::unique_ptr<opentracing::Span>& parentSpan)
+{
+    auto span = opentracing::Tracer::Global()->StartSpan(
+        "tracedSubroutine", { opentracing::ChildOf(&parentSpan->context()) });
+}
+
+void tracedFunction()
+{
+    auto span = opentracing::Tracer::Global()->StartSpan("tracedFunction");
+    tracedSubroutine(span);
+}
+
+}  // anonymous namespace
+
+int main(int argc, char* argv[])
+{
+    if (argc < 2) {
+        std::cerr << "usage: " << argv[0] << " <config-yaml-path>\n";
+        return 1;
+    }
+    setUpTracer(argv[1]);
```

Comment:
don't you need to close/flush the tracer before exit?

Reply:
Luckily, no. At least not for the Jaeger client. My favorite thing about C++ is that most things are auto-closing (formally called RAII). Spans must be auto-closing according to OT C++ spec, but the `Tracer::Close()` definition is a little more vague (see https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/tracer.h#L149-L153). As of now, we aren't using close any differently here than for spans. Might add it just in case I change my mind.

Reply:
Just to clarify, when the main function exits, the global tracer shared pointer is destroyed, invoking the Tracer destructor, which invokes `Tracer::close` on its own. We could manually invoke any of these methods, but there is no need to.

Reply:
Not saying you are wrong, but I'd expect that behavior if you had a locally scoped variable that would get destroyed upon exiting the main scope. That global variables behave the same way is a surprise to me.

Reply:
Ya it's a bit confusing. Here's a source quoting the standard about it: https://stackoverflow.com/a/2204628/1930331.